### PR TITLE
doc/gnome: make wrapGAppsHook4 more discoverable

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

A follow-up of https://github.com/NixOS/nixpkgs/pull/301865.

Also related: https://github.com/NixOS/nixpkgs/pull/227825

Previously wrapGAppsHook4 was sandwiched in the middle of a paragraph and only briefly described. I have restructured this paragraph. ~~In addition, I also mentioned gtk4-update-icon-cache to avoid potential problems caused by introducing gtk3 in the gtk4 package.~~

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: doc/languages-frameworks/gnome.section.md [hunk 1]
```diff
@@ -62,6 +62,18 @@ In the rare case you need to use icons from dependencies (e.g. when an app force
...
 
+It's also worth noting that some GTK4 packages rely on `gtk-update-icon-cache` to generate icon theme cache in Meson build scripts. They can be replaced with `gtk4-update-icon-cache` in GTK4 package, without adding GTK3 to `nativeBuildInputs`. This is usually something like:
```
Comment (jtojnar) [inline doc/languages-frameworks/gnome.section.md:65]: I am not fond of documenting downstream hacks for stuff that should be fixed upstream. Opened a tracking issue https://github.com/NixOS/nixpkgs/issues/302605

Comment (Aleksanaa) [reply doc/languages-frameworks/gnome.section.md:65]: Okay. Dropping this commit as of now.

**Comment by Aleksanaa** [comment]
@ofborg eval

**Comment by jtojnar** [comment]
Thinking about it a bit more, maybe we should rename the `wrapGAppsHook` to `wrapGAppsHook3`. That way, people will hopefully think twice about using it.

**Review by DanielSidhion — APPROVED** [review]
Given this is a very focused change in the docs to make it easier for users to understand what to use, I'm going ahead with merging it.

For the work mentioned to rename the function, I created an issue (already linked to this PR) so it can be done at an appropriate time.

I'm going ahead with merging this just so we don't block docs changes on that other work :)

**Pull request merged**